### ISSUE TITLE
not issue thank you for sharing code.

### ISSUE THREAD

**Issue by m0jek** [roccotsi2/esp32-smart-bms-simulation#2]
I live on a boat and have a battery bank of a several Stirling LiFePO4 batteries that include the Daly SmartBMS, but only accessible by Bluetooth. I can use the SmartBMS App, but I need to monitor all the time so I can warn if SOC drops too low etc.
Your simulator will hopefully help me test code so I don't have to sit in engine room all the time, while developing monitoring equipment. :-)

**Comment by roccotsi2** [comment]
Hi, thanks for your comment. :-)

It is nice to hear that my simulation is useful for someone else. I also do not have access to the "real" BMS, because my colleague has one. So I need a simulation for testing. By the way, you can also connect the SmartBMS app with my simulation (but only few operations are supported, mostly RunInfo as this hold the values that I am interested in).

Do you have the protocol specification for all of the SmartBMS functions?

**Comment by m0jek** [comment]
> 
> Do you have the protocol specification for all of the SmartBMS functions?

I had found some information on the protocol including a link to a data sheet on this forum thread;

https://diysolarforum.com/threads/decoding-the-daly-smartbms-protocol.21898/

Most people seem to be using the serial connectivity however, I have only access to SmartBMS Bluetooth as the LiFePO4 batteries I have, the BMS is built in but can only access using SmartBMS BLE.

Using the mobile phone app is no good as I would like the system to alert me.

**Comment by roccotsi2** [comment]
Unfortunately, the protocol mentioned in the link does not match the protocol my SmartBMS is using. I do not know why it is different (maybe a difference between serial and bluetooth connection), but it is different (e.g. the header my SmartBMS is using starts always with "D2 03").

Are you already using my simulation code on a ES32?

**Comment by m0jek** [comment]
Yes, yours was the only protocol that worked for me too. I'm using your simulation code successfully.

I have been struggling making progress until I found your code by chance. 

Will try again to reach out to Daly ... there must be complete information somewhere.

**Comment by roccotsi2** [comment]
If you get some official documentation, would be great if you share this with me.

If you need some additional information or function (beside RunInfo), please ask. If I get some time, I will try to analyze and add this.

**Comment by m0jek** [comment]
If I get something I will defiantly share with you. So far I have just silence ... :(

I hope to do some tests more this weekend and see if I can make some progress.

**Comment by m0jek** [comment]
> I hope to do some tests more this weekend and see if I can make some progress.

No progress yet, missed a weekend ...
